stdlib/public/core/Comparable.swift

```diff
@@ -218,3 +218,35 @@ extension Comparable {
     return !(lhs < rhs)
   }
 }
+
+extension Comparable { 
+  // Returns a value of type `Self` clamped to the provided `ClosedRange`.
+  //
+  // - Parameters:
+  //   - range: A value to determin the upper and lower bounds.
+  // - Returns: A value of type `Self` that is within the bounds of the provided `ClosedRange`.
+  @inlinable
+  func clamped(to range: ClosedRange<Self>) -> Self {
+    max(min(self, range.upperBound), range.lowerBound)
+  }
+  
+  // Returns a value of type `Self` clamped to the provided `PartialRangeFrom`.
+  //
+  // - Parameters:
+  //   - range: A value to determin the upper and lower bounds.
+  // - Returns: A value of type `Self` that is within the bounds of the provided `PartialRangeFrom`.
+  @inlinable
+  func clamped(to range: PartialRangeFrom<Self>) -> Self {
+    max(self, range.lowerBound)
+  }
+
+  // Returns a value of type `Self` clamped to the provided `PartialRangeThrough`
+  //
+  // - Parameters:
+  //   - range: A value to determin the upper and lower bounds
+  // - Returns: A value of type `Self` that is within the bounds of the provided `PartialRangeThrough`
+  @inlinable
+  func clamped(to range: PartialRangeThrough<Self>) -> Self {
+    min(self, range.upperBound)
+  }
+}
```

Comment:
Suggestions:

* Update copyright of this file to `2014 - 2020`.
* Use triple `///` comments, add example code, restrict width to 80 characters.
* Copy parameter names from [`Range.clamped(to:)`][] and [`ClosedRange.clamped(to:)`][].
* Add `@_alwaysEmitIntoClient` attribute and `public` keyword.
* TODO: Add tests to existing [Range.swift.gyb][] file?
* (Typo: `determin` → `determine`.)

[`Range.clamped(to:)`]:       <https://github.com/apple/swift/blob/8ec7bdf634586415a04939f5d1620b59a72873db/stdlib/public/core/Range.swift#L332-L364>
[`ClosedRange.clamped(to:)`]: <https://github.com/apple/swift/blob/8ec7bdf634586415a04939f5d1620b59a72873db/stdlib/public/core/ClosedRange.swift#L396-L428>
[Range.swift.gyb]:            <https://github.com/apple/swift/blob/8ec7bdf634586415a04939f5d1620b59a72873db/validation-test/stdlib/Range.swift.gyb#L515-L527>

```suggestion

extension Comparable {

  /// Returns a copy of this value, clamped to the given limiting range.
  ///
  ///     print(42.clamped(to: 0...100))
  ///     // Prints "42"
  ///
  ///     print(Int.min.clamped(to: 0...100))
  ///     // Prints "0"
  ///
  ///     print(Int.max.clamped(to: 0...100))
  ///     // Prints "100"
  ///
  /// - Parameter limits: The range to clamp this value.
  /// - Returns: A new value, clamped to the bounds of `limits`.
  @_alwaysEmitIntoClient
  public func clamped(to limits: ClosedRange<Self>) -> Self {
    max(min(self, limits.upperBound), limits.lowerBound)
  }

  /// Returns a copy of this value, clamped to the given limiting range.
  ///
  ///     print(42.clamped(to: 0...))
  ///     // Prints "42"
  ///
  ///     print(Int.min.clamped(to: 0...))
  ///     // Prints "0"
  ///
  ///     print(Int.max.clamped(to: 0...))
  ///     // Prints "9223372036854775807"
  ///
  /// - Parameter limits: The range to clamp this value.
  /// - Returns: A new value, clamped to the bounds of `limits`.
  @_alwaysEmitIntoClient
  public func clamped(to limits: PartialRangeFrom<Self>) -> Self {
    max(self, limits.lowerBound)
  }

  /// Returns a copy of this value, clamped to the given limiting range.
  ///
  ///     print(42.clamped(to: ...100))
  ///     // Prints "42"
  ///
  ///     print(Int.min.clamped(to: ...100))
  ///     // Prints "-9223372036854775808"
  ///
  ///     print(Int.max.clamped(to: ...100))
  ///     // Prints "100"
  ///
  /// - Parameter limits: The range to clamp this value.
  /// - Returns: A new value, clamped to the bounds of `limits`.
  @_alwaysEmitIntoClient
  public func clamped(to limits: PartialRangeThrough<Self>) -> Self {
    min(self, limits.upperBound)
  }
}
```